deepchem/feat/rdkit_grid_featurizer.py

```diff
@@ -467,38 +451,121 @@ def compute_pi_stack(protein_xyz,
             for each atom in ligand and in protein:
               add to list of atom indices
   """
-  raise ValueError("Not Yet Implemented")
-
-
-def is_cation_pi(cation_position, ring_center, ring_normal):
+  protein_pi_parallel = Counter()
+  protein_pi_t = Counter()
+  ligand_pi_parallel = Counter()
+  ligand_pi_t = Counter()
+
+  protein_aromatic_rings = []
+  ligand_aromatic_rings = []
+  for mol, ring_list in ((protein, protein_aromatic_rings),
+                         (ligand, ligand_aromatic_rings)):
+    aromatic_atoms = set(atom.GetIdx() for atom in mol.GetAromaticAtoms())
+    for ring in Chem.GetSymmSSSR(mol):
+      if set(ring).issubset(aromatic_atoms):
+        ring_center = compute_ring_center(mol, ring)
+        ring_normal = compute_ring_normal(mol, ring)
+        ring_list.append((ring, ring_center, ring_normal))
+
+  counted_pairs_parallel = set()
+  counted_pairs_t = set()
+  for prot_ring, prot_ring_center, prot_ring_normal in protein_aromatic_rings:
+    for lig_ring, lig_ring_center, lig_ring_normal in ligand_aromatic_rings:
+      if is_pi_parallel(
+          prot_ring_center,
+          prot_ring_normal,
+          lig_ring_center,
+          lig_ring_normal,
+          angle_cutoff=angle_cutoff,
+          dist_cutoff=dist_cutoff):
+        prot_to_update = set()
+        lig_to_update = set()
+        for i in prot_ring:
+          for j in lig_ring:
+            if (i, j) not in counted_pairs_parallel:
```

Comment:
Should we also be checking (j,i)?

Reply:
I don't think so. When we add tuples to the `counted_pairs_<parallel/t>`, the first index (`i`) is from the protein, and the second one (`j`) is from the ligand. So `(j, i)` is a different pair of atoms.
However, this points out that I should use more self-explanatory variable names here, it will improve readability.